### PR TITLE
Core metrics + development metrics dashboard

### LINKED ISSUE #3149: Adding micrometer metrics to Core engine
This ticket includes:
- [x] Adding tx timings and counters to the indexer
  - How fine grained should the metrics be? 
- [x] Adding query timings and counters
- [x] Adding `/promotheus` endpoint for scraping
- [x] Adding a simple Grafana dashboard from the exposed endpoint. This shouldn't be a production ready setup, but rather a dev setup that one can run to improve or work on the metrics reporting.

Open questions:
- Should the `promotheus` scrape endpoint sit together with the http module?
- Should there be some metric of size for a transactions ?

### PR DESCRIPTION
**Pull request by FiV0**
- This adds Timer metrics for queries + transactions
- Adds a simple development setup for prometheus + grafana that scrapes the metrics locally and displays them in a simple dashboard.

### CODE REVIEW

File: core/src/main/clojure/xtdb/indexer.clj
Comment (FiV0) [inline core/src/main/clojure/xtdb/indexer.clj]: @jarohen I don't know if all these metrics are bit overkill.

File: core/src/main/clojure/xtdb/indexer.clj
Comment (FiV0) [inline core/src/main/clojure/xtdb/indexer.clj]: @jarohen As `call` might be recursive, this might double counting things.

File: core/src/main/clojure/xtdb/indexer.clj
Comment (jarohen) [inline core/src/main/clojure/xtdb/indexer.clj]: I'd probably have these as a map, save adding a field every time we want to add a different timer?

File: core/src/main/clojure/xtdb/indexer.clj
Comment (jarohen) [reply core/src/main/clojure/xtdb/indexer.clj]: yeah :+1: 100% accuracy probably isn't required here, so not too worried about that tbh. 

To your point about granularity, now that you have the wiring set up it might be a good idea to talk about what metrics you think we should have?

File: core/src/main/clojure/xtdb/indexer.clj
Comment (jarohen) [reply core/src/main/clojure/xtdb/indexer.clj]: yeah, that'd be my gut feel too tbh. shall we maybe start with one for the tx as a whole, one for the queries, and go from there?

File: core/src/main/clojure/xtdb/metrics.clj
Comment (jarohen) [inline core/src/main/clojure/xtdb/metrics.clj]: `doseq` is probably your friend here :slightly_smiling_face: :golf:

File: core/src/main/clojure/xtdb/metrics.clj
Comment (jarohen) [inline core/src/main/clojure/xtdb/metrics.clj]: do we need to specify these? I think any value we put here will be both arbitrary and likely inaccurate, so would err on the side of not providing them at all unless it has other impacts I'm not aware of

File: modules/bench/README.adoc
Comment (jarohen) [inline modules/bench/README.adoc]: nit: adoc idiom is [one sentence per line](https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line), makes git diffs easier to read/work with in future.

another nit: aware it's internal docs but worth giving them a bit more structure - this reads a little like a brain dump. have a think on what the reader will need to extract from this :slightly_smiling_face:

**Review by jarohen — CHANGES_REQUESTED** [review]
Cheers @FiV0 :pray: looking forward to a demo with some shiny graphs in the show-n-tell tomorrow :slightly_smiling_face:

File: core/src/main/clojure/xtdb/indexer.clj
Comment (FiV0) [reply core/src/main/clojure/xtdb/indexer.clj]: Yes agree. As this is likely some hot loop I wanted to avoid the extra lookup every time a transaction gets indexed.

File: modules/bench/README.adoc
Comment (FiV0) [reply modules/bench/README.adoc]: Reworded a bit.

File: core/src/main/clojure/xtdb/indexer.clj
Comment (jarohen) [inline core/src/main/clojure/xtdb/indexer.clj]: nit: indent

File: core/src/main/clojure/xtdb/indexer.clj
Comment (jarohen) [inline core/src/main/clojure/xtdb/indexer.clj]: use `^Map` here - if the map's small enough, you're not guaranteed to get a PHM in Clojure.

(if you even need the type-hint at all, that is)

File: core/src/main/clojure/xtdb/indexer.clj
Comment (jarohen) [inline core/src/main/clojure/xtdb/indexer.clj:577]: (not one for now, your call)

When we write a few more of these, I might be tempted by a helper macro:

```clojure
(metrics/with-timer (:tx-timer metrics)
  (case (.getLeg ...)
    ...))
```

**Review by jarohen — APPROVED** [review]
cheers @FiV0!

marked approved, so don't worry about a re-review before merging :heavy_check_mark:

**Pull request merged**
